livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -378,13 +387,49 @@ private void cancelStatement(int statementId) throws LivyException {
     callRestAPI("/sessions/" + sessionInfo.id + "/statements/" + statementId + "/cancel", "POST");
   }
 
-  private RestTemplate getRestTemplate() {
+
+  private RestTemplate getRestTemplate() throws LivyException {
+    HttpClient httpClient = null;
+    if (livyURL.startsWith("https:")) {
+      String keystoreFile = property.getProperty("zeppelin.livy.ssl.trustStore");
+      String password = property.getProperty("zeppelin.livy.ssl.trustStorePassword");
```

Comment:
Fixed, and also make another improvement to create `RestTemplate` in constructor. Otherwise each livy rest api would construct a new `RestTemplate`.